File: core/src/main/java/vn/loitp/function/notification/NottiFactory.java
package vn.loitp.function.notification;

import vn.loitp.function.notification.notifications.BIgTextNotification;
import vn.loitp.function.notification.notifications.BigPictureNotification;
import vn.loitp.function.notification.notifications.CustomNotification;
import vn.loitp.function.notification.notifications.InboxNotification;
import vn.loitp.function.notification.notifications.StandardNotification;

public class NottiFactory {

    public enum TYPE {INBOX, BIG_TEXT, BIG_PICTURE, STANDARD}

    public static CustomNotification get(TYPE type, String title, String content) {
        switch (type) {
            case STANDARD:
                return new StandardNotification(title, content);
            case BIG_TEXT:
                return new BIgTextNotification(title, content);
            case BIG_PICTURE:
                return new BigPictureNotification(title, content);
            case INBOX:
                return new InboxNotification(title, content);
            default:
                return new StandardNotification(title, content);
        }
    }
}

File: core/src/main/java/com/loitpcore/function/notification/NottiFactory.kt
package com.loitpcore.function.notification

import com.loitpcore.function.notification.notifications.*

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
object NottiFactory {

    enum class TYPE {
        INBOX,
        BIG_TEXT,
        BIG_PICTURE,
        STANDARD
    }

    operator fun get(
        type: TYPE?,
        title: String?,
        content: String?
    ): CustomNotification {

        return when (type) {
            TYPE.STANDARD -> {
                StandardNotification(title, content)
            }
            TYPE.BIG_TEXT -> {
                BIgTextNotification(title, content)
            }
            TYPE.BIG_PICTURE -> {
                BigPictureNotification(title, content)
            }
            TYPE.INBOX -> {
                InboxNotification(title, content)
            }
            else -> {
                StandardNotification(title, content)
            }
        }
    }
}
